app/views/layouts/application.html.erb

```diff
@@ -1,5 +1,6 @@
 <%= content_for :navbar do %>
   <%= hidden_field_tag("mu-current-exp", UserStats.exp_for(@current_user)) if in_gamified_context? %>
+  <%= hidden_field_tag("mu-uppercase-mode-enabled", @current_user.uppercase_mode_enabled) %>
```

Comment:
I think this is fine right now. However, take into consideration that there may be a few more _preferences_ , so extracting a specific partial may be a good refactor in the future. 

I am even wondering if it would be good idea to have an embedded `preferences` object at user just for sake of keeping things organized. Something like this: `user.preferences.uppercase?`

Reply:
Other preferences should be: 

```ruby
user.preferences.gamified?
user.preferences.emojiless?
user.preferences.dark?
```

Reply:
Just like `Mumuki::Domain::Organization::Settings`?

There's an answer I read some time ago which I liked a lot: https://stackoverflow.com/a/24456817 I think the unobtrusive JS or the gon gem could both work.

We already have a lot of `hidden_field_tag`s spread out through the views, such as for cache enabling/disabling or on the exercise view https://github.com/mumuki/mumuki-laboratory/blob/92165951487bdc56472fbd94fff669595c9fe0b5/app/views/exercises/show.html.erb#L47-L50

so I'm sure a little more structure -  either through a data object or using gon - would be of great use. If we're not changing the existing code for now, let's at least make the new code cleaner

Reply:
No, not really like Settings, but like this https://github.com/mumuki/mumuki-laboratory/issues/1527#issuecomment-749899337

Reply:
Or maybe yes... let me explain a bit better: 

1. Yes, I like the `preferences.json` approach used in the exercise's settings at the front-end, so in that sense, I agree with your proposal of using it  - or some even better strategy - and having a preferences object in the JS world
2. In the backend, I'm not talking about using a serialization strategy like with organization's settings, but like the assignment, that uses embeddables - `composed_of`. That means not storing a JSON at the database which has proven to be cumbersome, but plain old columns that are still refied at the OOP level.